scripts/coccinelle/boolean-expression.cocci

```diff
@@ -0,0 +1,357 @@
+/// Make statements evaluate Boolean expressions
+///
+// Confidence: Low
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+
+@depends on !(file in "ext")@
+expression *X;
+statement S;
+@@
+
+- if (X)
++ if (X != NULL)
+  S
+
+@depends on !(file in "ext")@
+expression *X;
+statement S1, S2;
+@@
+
+- if (X) {
++ if (X != NULL) {
+S1
+} else {
+S2
+}
+
+@depends on !(file in "ext")@
+expression *X;
+statement S;
+@@
+
+- if (X) {
++ if (X != NULL) {
+S
+}
+
+@depends on !(file in "ext")@
+typedef uint8_t;
+typedef u8_t;
+typedef uint16_t;
+typedef u16_t;
+typedef uint32_t;
+typedef u32_t;
+{uint8_t,u8_t,uint16_t,u16_t,uint32_t,u32_t,unsigned int,int,size_t,unsigned} I;
+statement S;
+@@
+
+- if (!I)
++ if (I == 0)
+S
+
+@depends on !(file in "ext")@
+expression *E;
+@@
+
+  E ==
+- 0
++ NULL
+
+@depends on !(file in "ext")@
+expression *E;
+@@
+
+- 0
++ NULL
+  == E
+
+@depends on !(file in "ext")@
+expression *E;
+@@
+
+  E !=
+- 0
++ NULL
+
+@depends on !(file in "ext")@
+expression *E;
+@@
+
+- 0
++ NULL
+  != E
+
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (X && E)
++ if ((X != NULL) && E)
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (!X && E)
++ if ((X == NULL) && E)
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (E && X)
++ if (E && (X != NULL))
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (!X && E)
++ if (E && (X == NULL))
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+statement S;
+@@
+
+- if (!X)
++ if (X == NULL)
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (X || E)
++ if ((X != NULL) || E)
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (!X || E)
++ if ((X == NULL) || E)
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (E || X)
++ if (E || (X != NULL))
+  S
+
+@depends on !(file in "ext")@
+idexpression *X;
+expression E;
+statement S;
+@@
+
+- if (!X || E)
++ if (E || (X == NULL))
+  S
+
+@depends on !(file in "ext")@
+statement S;
+expression E;
+constant C;
+@@
+
+- if (E & C)
++ if ((E & C) != 0)
+  S
+
+@depends on !(file in "ext")@
+statement S;
+identifier I;
+constant C;
+@@
+
+- if (!(I & C))
++ if ((I & C) == 0)
+  S
+
+
+@depends on !(file in "ext")@
+statement S;
+expression E;
+constant C;
+@@
+
+- if (C & E)
++ if ((C & E) != 0)
+  S
+
+@depends on !(file in "ext")@
+statement S;
+identifier I;
+constant C;
+@@
+
+- if (I & C)
```

Comment:
Any particular reson for using `identifier` and `constant`.
I assume your main intention is to change
```
- if (A & B)
+ if ((A & B) != 0)
```
without caring about the type of A & B.
So, I think both A, B can be simply `expression` metavariables.